validator/rpc/handlers_health.go

```diff
@@ -0,0 +1,165 @@
+		return
+	}
+
+	http2.WriteJson(w, struct {
+		Beacon    string `json:"beacon"`
+		Validator string `json:"validator"`
+	}{
+		Beacon:    beacon.Version,
+		Validator: version.Version(),
+	})
+}
+
+// StreamBeaconLogs from the beacon node via a gRPC server-side stream.
+func (s *Server) StreamBeaconLogs(w http.ResponseWriter, r *http.Request) {
+	// Wrap service context with a cancel in order to propagate the exiting of
+	// this method properly to the beacon node server.
+	ctx, span := trace.StartSpan(r.Context(), "validator.web.health.StreamBeaconLogs")
+	defer span.End()
+	// Set up SSE response headers
+	w.Header().Set("Content-Type", "text/event-stream")
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Connection", "keep-alive")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+
+	// Flush helper function to ensure data is sent to client
+	flusher, ok := w.(http.Flusher)
+	if !ok {
+		http2.HandleError(w, "Streaming unsupported!", http.StatusInternalServerError)
+		return
+	}
+	// TODO: StreamBeaconLogs grpc will need to be replaced in the future
+	client, err := s.beaconNodeHealthClient.StreamBeaconLogs(ctx, &emptypb.Empty{})
+	if err != nil {
+		http2.HandleError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	for {
+		select {
+		case <-s.ctx.Done():
+			return
+		case <-ctx.Done():
+			return
+		case <-client.Context().Done():
+			return
+		default:
+			logResp, err := client.Recv()
+			if err != nil {
+				http2.HandleError(w, "could not receive beacon logs from stream", http.StatusInternalServerError)
+				return
+			}
+			jsonResp, err := json.Marshal(logResp)
+			if err != nil {
+				http2.HandleError(w, "could not encode long response", http.StatusInternalServerError)
+				return
+			}
+
+			// Send the response as an SSE event
+			// Assuming resp has a String() method for simplicity
+			_, err = fmt.Fprintf(w, "%s\n", jsonResp)
+			if err != nil {
+				http2.HandleError(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			// Flush the data to the client immediately
+			flusher.Flush()
+		}
+	}
+}
+
+// StreamValidatorLogs from the validator client via a gRPC server-side stream.
+func (s *Server) StreamValidatorLogs(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.web.health.StreamValidatorLogs")
+	defer span.End()
+
+	// Ensure that the writer supports flushing.
+	flusher, ok := w.(http.Flusher)
+	if !ok {
+		http2.HandleError(w, "Streaming unsupported!", http.StatusInternalServerError)
+		return
+	}
+
+	ch := make(chan []byte, s.streamLogsBufferSize)
+	sub := s.logsStreamer.LogsFeed().Subscribe(ch)
+	defer func() {
+		sub.Unsubscribe()
+		close(ch)
+	}()
+	// Set up SSE response headers
+	w.Header().Set("Content-Type", "text/event-stream")
+	w.Header().Set("Cache-Control", "no-cache")
+	w.Header().Set("Connection", "keep-alive")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
Is this a potential security issue? Should be configurable, or I believe use the cors of the overall server

Reply:
removing the header for now, the cors middleware will be addressed here https://github.com/prysmaticlabs/prysm/issues/12958